From 19945ae59bc56ebf2be10a440695cacf31fc3391 Mon Sep 17 00:00:00 2001
From: Dai Xin <daixin@wxiat.com>
Date: Tue, 1 Nov 2022 17:07:27 +0000
Subject: [PATCH] sw64: acpi: fix compilation dependency when CONFIG_PCI=n
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 19945ae59bc56ebf2be10a440695cacf31fc3391
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56OLG

--------------------------------

Signed-off-by: Dai Xin <daixin@wxiat.com>
Reviewed-by: He Sheng <hesheng@wxiat.com>
Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/hwmon/sw64_pvt.c           | 1 +
 drivers/irqchip/irq-sw64-intc-v2.c | 4 ++++
 2 files changed, 5 insertions(+)

diff --git a/drivers/hwmon/sw64_pvt.c b/drivers/hwmon/sw64_pvt.c
index 9e292a90af38..29098bc1f645 100644
--- a/drivers/hwmon/sw64_pvt.c
+++ b/drivers/hwmon/sw64_pvt.c
@@ -11,6 +11,7 @@
 #include <linux/platform_device.h>
 #include <linux/hwmon.h>
 #include <linux/hwmon-sysfs.h>
+#include <linux/io.h>
 #include <linux/module.h>
 
 #define PVT_VSYS                      0
diff --git a/drivers/irqchip/irq-sw64-intc-v2.c b/drivers/irqchip/irq-sw64-intc-v2.c
index 8640c4aa9506..070c12ede959 100644
--- a/drivers/irqchip/irq-sw64-intc-v2.c
+++ b/drivers/irqchip/irq-sw64-intc-v2.c
@@ -34,6 +34,7 @@ static const struct irq_domain_ops sw64_intc_domain_ops = {
 	.map = sw64_intc_domain_map,
 };
 
+#ifdef CONFIG_ACPI
 static int __init
 intc_parse_madt(union acpi_subtable_headers *header,
 		       const unsigned long end)
@@ -91,7 +92,9 @@ static int __init intc_init(void)
 }
 
 subsys_initcall(intc_init);
+#endif
 
+#ifdef CONFIG_OF
 static struct irq_domain *root_domain;
 
 static int __init
@@ -143,3 +146,4 @@ init_onchip_vt_IRQ(struct device_node *intc, struct device_node *parent)
 }
 
 IRQCHIP_DECLARE(sw64_vt_intc, "sw64,sw6_irq_vt_controller", init_onchip_vt_IRQ);
+#endif
-- 
2.33.0

